api/v1alpha1/workflownode_types.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName=wfn
+type WorkflowNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a pod chaos experiment
+	Spec WorkflowNodeSpec `json:"spec"`
+
+	// +optional
+	// Most recently observed status of the chaos experiment about pods
+	Status WorkflowNodeStatus `json:"status"`
+}
+
+type WorkflowNodeSpec struct {
+	TemplateName string       `json:"template_name"`
+	WorkflowName string       `json:"workflow_name"`
+	Type         TemplateType `json:"type"`
+	StartTime    *metav1.Time `json:"start_time"`
+	// +optional
+	Deadline *metav1.Time `json:"deadline,omitempty"`
+	// +optional
+	Tasks []string `json:"tasks,omitempty"`
+	// +optional
+	*EmbedChaos `json:",inline,omitempty"`
+}
+
+type WorkflowNodeStatus struct {
+
+	// ExpectedChildren means the expected children to execute
+	// +optional
+	ExpectedChildren *int `json:"expected_children,omitempty"`
+
+	// ChaosResource refs to the real chaos CR object.
+	// +optional
+	ChaosResource *corev1.TypedLocalObjectReference `json:"chaos_resource,omitempty"`
+
+	// ActiveChildren means the created children node
+	// +optional
+	ActiveChildren []corev1.LocalObjectReference `json:"active_children,omitempty"`
+
+	// Children is necessary for representing the order when replicated child template references by parent template.
+	// +optional
+	FinishedChildren []corev1.LocalObjectReference `json:"finished_children,omitempty"`
```

Comment:
Could you explain the function of `FinishedChildren` a bit more in detail pls? I don't quite get it.

Reply:
This field refs to the children nodes which accomplished. For example, One serial node called A has three children nodes: B, C, D; when B and C finished, D is running, the status of A just like:

```jsonc
{
"expected_children": 3,
"active_children": ["D"],
"finished_children": ["B", "C"],
}
```